### ISSUE TITLE
The word suggestion become blue (highlighted) when scrolling them with DPAD LEFT and DPAD RIGHT

### ISSUE THREAD

**Issue by chaimsha** [sspanak/tt9#640]
There is a problem that the color turns blue when you go to continue choosing the right word. That is, when you go to choose a word using the arrows and it goes to the continuation of the words that are displayed on the screen, then everything is blue and it is impossible to see exactly where you stand.
A video is attached

Thank you very much I really like this keyboard and I appreciate your good investment.

Uploading T9.mp4…

**Comment by chaimsha** [comment]
https://github.com/user-attachments/assets/db094dc2-5f31-450b-a1f6-27c799036ee3

**Comment by sspanak** [comment]
This looks like a duplicate of #478. Are you also using a TIQ M5?

Either way, I have no idea why this happens and how to fix it, unfortunately. I can't reproduce it on any of my testing devices.

**Comment by chaimsha** [comment]
hi
I attached here according to what I took from one of your instructions, but I didn't see anything written LOG. Is it enough?
Thank you very much this will help a lot

https://github.com/user-attachments/assets/68480d65-9960-4bc9-9944-660217d9a8fc

**Comment by sspanak** [comment]
No, sorry, it doesn't work this way. The minimum useful information you can provide is the phone make and model. Additionally, it would be great if you can tell if the problem appears or goes away in some application(s), and, if possible, describe the steps to consistently reproduce the problem.

**Comment by yh26vh26** [comment]
Maybe it happens because it's Android 8.
 It also happens to me on this phone that I am attaching the link here.
https://www.jethroshop.com/phone
It would be very helpful if it is possible to fix it.
Thank you very much

**Comment by sspanak** [comment]
I don't think so. The same happens on TIQ M5 which runs Android 12, if I remember correctly. On the other hand, I have an Android 8.1 phone on which the problem has never occurred.

**Comment by AshiVered** [comment]
His phone is called Show F2, if I recognize correctly.
A niche device sold only in Israel, running Android 8.1. If I understood correctly, the problem with all apps.

Don't know if it will help you solve the problem, but in QinBoard T9 this does not happen in my opinion

**Comment by sspanak** [comment]
I am sure it happens on no other keyboard. It is a weird problem that happens for no reason.

I see this issue is getting more active, so I'll ask a couple of questions that may help me understand better what is going on (no promises though).

1. Does the problem occur after pressing a key, or TT9 just appears in blue from the start?
2. If it happens only after pressing a key, does it happen when you scroll a suggestion, or when you press LEFT or RIGHT to scroll, when there are NO suggestions? Or does it happen when you press some other key?
3. If your phone has available unused keys when typing, like MENU or F1-F4, or others, you can try reassigning the scrolling functions to those keys. Go to Settings -> Keypad -> Hotkeys and find Previous and Next Suggestion. Make sure to unassign the DPAD arrow keys and select something else. Does the problem still occur when not using the DPAD to scroll the suggestions?
4. Does the problem go away, if you go to the home screen, then to the last application you were using?
5. Does the problem go away when you go to another application?
6. Does the problem go away if you change the layout (Settings -> Appearance -> Screen Layout)? Try switching to "Invisible" or "Numpad", then back to the one you were using in the first place.

As a technical summary, I suspect the problem may be related to returning the incorrect value when handling the DPAD keys. I remember that on some particular Kyocera phone, I had to call super.onKeyDown()/super.onKeyUp() when not handling a key, otherwise some keys were misbehaving. On the other hand, other phones require simply returning "false".

**Comment by chaimsha** [comment]
I checked it,
1. The status line of the language like English, ABC - it's immediately in English.
2. But in the line of words, when you tap a box and scroll, then the first word always remains in blue and the other word where you are standing is only slightly highlighted, and when you reach the end and move on, then everything turns blue and you don't see where you are standing at all.
This happens when alternate word scrolling is set. If not, it doesn't happen that everything is blue, but then there are other problems that only one box is blue and besides that as appears in number #648

3. I tried it and it didn't help.
4. Didn't help either.
5. Didn't help either.
6. Didn't help either.

Thank you very much

**Comment by sspanak** [comment]
Now I understand the problem even less than before...

OK, here is one more scenario to try:
1. Restart the phone, then go somewhere you can type, but do not press any key. Is everything blue again?
2. If it wasn't blue, press a key to type one letter. Did the list of letter come up in blue?
3. If it still isn't blue, scroll the list and look for the exact moment when it becomes blue.

The problem may also be related to focusing, like @wspdyb said in #478, but I am not sure how. I never set the focus to anything.

**Comment by sspanak** [comment]
Everyone, could you please try the new version 41.0? It fixes a very similar problem on CAT S22, so it may fix it for you, too.

**Comment by sspanak** [comment]
Excellent! I'll keep this open for a short while, then I'll consider the problem to be solved and close it.
